pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -74,23 +74,32 @@ const (
 
 const (
 	cephOSDStart = `
-function sigterm() {
-	echo "SIGTERM received"
+RESTART_INTERVAL=%d
+child_pid=""
+
+function handle_signal() {
+	echo "signal received"
+	kill -TERM "$child_pid"
 	exit
 }
 
-trap sigterm SIGTERM
+trap handle_signal TERM KILL
```

Comment:
This is fine. According to this article, I think if bash is sigkill'ed, the children will also be sigkilled because bash is pid 1. https://dev.betterdoc.org/docker/linux/container/signals/pid1/2021/06/18/how-docker-forced-me-to-learn-more-about-linux.html